pkg/restore/restore.go

```diff
@@ -259,50 +259,98 @@ func (ctx *context) execute() (api.RestoreResult, api.RestoreResult) {
 func (ctx *context) restoreFromDir(dir string) (api.RestoreResult, api.RestoreResult) {
 	warnings, errors := api.RestoreResult{}, api.RestoreResult{}
 
-	// cluster-scoped
-	clusterPath := path.Join(dir, api.ClusterScopedDir)
-	exists, err := ctx.fileSystem.DirExists(clusterPath)
-	if err != nil {
-		errors.Cluster = []string{err.Error()}
-	}
-	if exists {
-		w, e := ctx.restoreNamespace("", clusterPath)
-		merge(&warnings, &w)
-		merge(&errors, &e)
-	}
+	namespaceFilter := collections.NewIncludesExcludes().Includes(ctx.restore.Spec.IncludedNamespaces...).Excludes(ctx.restore.Spec.ExcludedNamespaces...)
 
-	// namespace-scoped
-	namespacesPath := path.Join(dir, api.NamespaceScopedDir)
-	exists, err = ctx.fileSystem.DirExists(namespacesPath)
+	// Make sure the top level "resources" dir exists:
+	resourcesDir := path.Join(dir, api.ResourcesDir)
+	_, err := ctx.fileSystem.DirExists(resourcesDir)
 	if err != nil {
 		addArkError(&errors, err)
 		return warnings, errors
 	}
-	if !exists {
-		return warnings, errors
-	}
 
-	nses, err := ctx.fileSystem.ReadDir(namespacesPath)
+	resourceDirs, err := ctx.fileSystem.ReadDir(resourcesDir)
```

Comment:
Do you think there's any performance advantage one way or another between reading the resources dir up front like this, and just calling `DirExists` on each `resource` from `ctx.prioritizedResources`?

Reply:
I can't come up with a theory one way or the other, would have to test but in context of restore my guess would be not significant at this point. (backup might be another story as I assume people might be running that a lot more)

If worthwhile let me know and I'll test next week.